hyperspy/signal.py

```diff
@@ -4005,38 +4011,68 @@ def is_rgb(self):
     def is_rgbx(self):
         return rgb_tools.is_rgbx(self.data)
 
-    def add_marker(self, marker, plot_on_signal=True, plot_marker=True):
+    def add_marker(
+            self, marker, plot_on_signal=True, plot_marker=True,
+            permanent=False):
         """
         Add a marker to the signal or navigator plot.
 
         Plot the signal, if not yet plotted
 
         Parameters
         ----------
-        marker: `hyperspy.drawing._markers`
-            the marker to add. see `plot.markers`
-        plot_on_signal: bool
+        marker : `hyperspy.drawing._markers`
+            the marker to add. See `plot.markers`
+        plot_on_signal : bool
             If True, add the marker to the signal
             If False, add the marker to the navigator
-        plot_marker: bool
+        plot_marker : bool
             if True, plot the marker
+        permanent : bool, default False
+            If False, the marker will only appear in the current
+            plot. If True, the marker will be added to the
+            markers list, and be plotted with plot(plot_markers=True).
 
         Examples
         -------
         >>> import scipy.misc
         >>> im = hs.signals.Signal2D(scipy.misc.ascent())
-        >>> m = hs.plot.markers.rectangle(x1=150, y1=100, x2=400,
+        >>> m = hs.markers.rectangle(x1=150, y1=100, x2=400,
         >>>                                  y2=400, color='red')
         >>> im.add_marker(m)
 
+        Add permanent marker
+        >>> s = hs.signals.Signal2D(np.random.random(100, 100))
+        >>> marker = hs.markers.point(50, 60)
+        >>> s.add_marker(marker, permanent=True, plot_marker=False)
+        >>> s.plot(plot_markers=True) #doctest: +SKIP
+
+        Add permanent marker which changes with navigation position
+        >>> s = hs.signals.Signal2D(np.random.randint(10, size=(3, 100, 100)))
+        >>> marker = hs.markers.point((10, 30, 50), (30, 50, 60), color='red')
+        >>> s.add_marker(marker, permanent=True, plot_marker=False)
+        >>> s.plot(plot_markers=True) #doctest: +SKIP
         """
-        if self._plot is None:
-            self.plot()
-        if plot_on_signal:
-            self._plot.signal_plot.add_marker(marker)
-        else:
-            self._plot.navigator_plot.add_marker(marker)
+        marker._plot_on_signal = plot_on_signal
         if plot_marker:
+            if self._plot is None:
+                self.plot()
+            if marker._plot_on_signal:
+                self._plot.signal_plot.add_marker(marker)
+            else:
+                self._plot.navigator_plot.add_marker(marker)
+            marker.plot()
+        if permanent:
+            if not hasattr(self, "markers"):
+                self.markers = []
+            self.markers.append(marker)
```

Comment:
I would argue that previous implementation is a little unintuitive from a HyperSpy perspective. It seems the previous implementation is very MPL-centric. As far as I understand it, `add_marker` previously:
 - associated the marker with a specific MPL axes.
 - `marker.plot()` then plotted the marker on the associated axes object.
 - if the signal was replotted, marker kindof broke down (?).

A more intuitive approach can be added here:
 - always have `add_marker` plot the marker onto the axes (no matter the args), but maybe not cause the figure to be drawn (determined by arg?). This last point would allow for adding several markers to the figure without flicker, but would still not give the speedup of collections.
 - have `add_marker` add the marker to the signal (like you have done) when permanent